src/marqo/tensor_search/tensor_search.py

```diff
@@ -165,17 +174,20 @@ def _add_documents_unstructured(config: Config, add_docs_params: AddDocsParams,
                 tensor_fields_and_multimodal_subfields = copy.deepcopy(add_docs_params.tensor_fields) \
                     if add_docs_params.tensor_fields else []
                 tensor_fields_and_multimodal_subfields.extend(multimodal_sub_fields)
-                image_repo = exit_stack.enter_context(
-                    add_docs.download_and_preprocess_images(
-                        docs=docs, thread_count=add_docs_params.image_download_thread_count,
+                content_repo = exit_stack.enter_context(
+                    add_docs.download_and_preprocess_content(
+                        docs=docs,
+                        thread_count=media_download_thread_count,
```

Comment:
Yeah, we only use one thread count variable. I don't think we can have different thread counts for different modalities. For example a batch has documents containing 3 modalities at the same time. To make different thread counts work, we'd have to split up the documents into categories, then multithread download each modality group one after the other with the corresponding number of threads the user specified for that modality, or perhaps mini group them and dynamically change the number of threads as different modalities come in. Can you think of a better way to implement this idea?

This would require quite a bit of refactoring of our code in this file. Is it super necessary at this point? Perhaps we can do this in a future release?